### ISSUE TITLE
Question: Will `amqp-client.js` Work on React Native Based Mobile App?

### ISSUE THREAD

**Issue by mcgill-sfeng** [cloudamqp/amqp-client.js#107]
Will `amqp-client.js` work on react native based mobile app?

When I'm looking at [RabbitMQ's documentation](https://www.rabbitmq.com/docs/devtools) they gave a separate section for the client to be used for Android and iOS, which is built with Swift and Objective-C.

If I want to build a mobile app with React native in JS and I need an AMQP client, any chance `amqp-client.js` can help me on that? If not, what would be my best alternatives?

**Comment by carlhoerberg** [comment]
Yes! Works well if you put the [websocket proxy](
https://github.com/cloudamqp/amqproxy) before rabbitmq (we install it by
default on all CloudAMQP servers). LavinMQ has websocket support built in.

**Comment by mcgill-sfeng** [comment]
@carlhoerberg thanks for the prompt response!

So if I'm understanding it correctly, the recommended way is to use `amqp-client.js` on the React Native client app (which is by default connected to `CloudAMQP`), and install and use `LavinMQ` on the server, right? Just to confirm that for this approach I wouldn't need to install extra instances of RabbitMQ/LavinMQ on the client app, which is what I'm trying to avoid here.

**Comment by dentarg** [comment]
@mcgill-sfeng Yes! Pick a [CloudAMQP plan with LavinMQ](https://www.cloudamqp.com/plans.html) and you should be good.

Just to clarify, the websocket proxy is this project https://github.com/cloudamqp/websocket-tcp-relay

**Comment by dentarg** [comment]
@mcgill-sfeng just to be extra clear, all plans on CloudAMQP has websocket support out of the box because we run the proxy for you if you go with RabbitMQ (but for LavinMQ it isn't needed at all)
